pkg/services/ngalert/schedule/schedule.go

```diff
@@ -221,17 +230,41 @@ func (sch *schedule) SyncAndApplyConfigFromDatabase() error {
 
 		existing, ok := sch.senders[cfg.OrgID]
 
-		// We have no running sender and no Alertmanager(s) configured, no-op.
-		if !ok && len(cfg.Alertmanagers) == 0 {
-			sch.log.Debug("no external alertmanagers configured", "org", cfg.OrgID)
-			continue
-		}
 		//  We have no running sender and alerts are handled internally, no-op.
 		if !ok && cfg.SendAlertsTo == ngmodels.InternalAlertmanager {
 			sch.log.Debug("alerts are handled internally", "org", cfg.OrgID)
 			continue
 		}
 
+		// We might have alertmanager datasources that are acting as external
+		// alertmanager, let's fetch them.
+		query := &datasources.GetDataSourcesByTypeQuery{
+			OrgId: cfg.OrgID,
+			Type:  "alertmanager",
+		}
+		err = sch.datasourceService.GetDataSourcesByType(context.Background(), query)
```

Comment:
We need to be careful here again as since we have no timeout we can lock `adminConfigMtx` for an extended duration of time.

Reply:
Where is the timeout, I'm not sure I can see it?

Reply:
I think 5 or even 15 seconds is OK, we just don't want to hold the lock too long if the query either takes a long time or the response from the server takes a long time (i.e. lots of packet loss, corruption and retransmission, etc). It's weird that `SyncAndApplyConfigFromDatabase()` does not take a context.